vllm/transformers_utils/config.py

```diff
@@ -104,6 +110,56 @@ def get_config(
     return config
 
 
+def load_params_config(model, revision) -> PretrainedConfig:
```

Comment:
Is there a reason why you have this new config? It seems to have the same information you would have in the config.json, just named differently

Reply:
The main reason is because the original format is always stored in [params.json](https://huggingface.co/mistralai/Mistral-7B-Instruct-v0.3/blob/main/params.json) which accompanies the consolidated.safetensors checkpoints.

Guess there are two problems with config.json

- 1. They are tied quite heavily to transformers. So if one wants to add a new, no transformers-formatted model it doesn't make too much sense to follow transformers config style
- 2. They tend to be quite bloated, e.g. has info that's outdated such as `sliding_window` here: https://huggingface.co/mistralai/Mistral-7B-Instruct-v0.3/blob/e0bc86c23ce5aae1db576c8cca6f06f1f73af2db/config.json#L19